### PR TITLE
[New module] Freesurfer synthregistration

### PR DESCRIPTION
**Pull request by anroy1**
## Describe your changes

Initial module PR for registration tool using Freesurfer synthmorph

## Checklist before requesting a review

- Create the tool:
  - [x] Edit `./modules/nf-scil/<category>/<tool>/main.nf`
  - [x] Edit `./modules/nf-scil/<category>/<tool>/meta.yml`
- Generate the tests:
  - [x] Edit `./tests/modules/nf-scil/<category>/<tool>/main.nf`
  - [x] Edit `./tests/modules/nf-scil/<category>/<tool>/nextflow.config`
  - [x] Add test data locally for tests with the fork repository
  - [x] Generate the test infrastructure and _md5sum_ for all outputs
- Ensure the syntax is correct :
  - [x] Check indentation abides with the rest of the library (don't hesitate to correct others !)
  - [x] Lint everything. Ensure your variables have good names.

### CODE REVIEW

File: modules/nf-neuro/registration/synthregistration/environment.yml
Comment (AlexVCaron) [inline modules/nf-neuro/registration/synthregistration/environment.yml:1]: As stated in #5

File: modules/nf-neuro/registration/synthregistration/main.nf
Comment (AlexVCaron) [inline modules/nf-neuro/registration/synthregistration/main.nf:1]: As discussed in #5 , let's decide on a naming convention for the affine/linear part and drop `init` when talking about it.

File: modules/nf-neuro/registration/synthregistration/meta.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+name: "registration_synthregistration"
+description: Perform registration using SynthMorph from Freesurfer. Outputs transforms in Freesurfer format .lta for affine and .nii.gz (synthmorph also supports .mgz) for deform, both in RAS orientation, which can be converted for ANTs with respectively lta_convert and mri_warp_convert, which support a wide range of conversion formats and orientations. Conversion can be processed using the registration/convert module which can be used successively to this one.
```
Comment (AlexVCaron) [inline modules/nf-neuro/registration/synthregistration/meta.yml:3]:
```suggestion
description: Perform registration using SynthMorph from Freesurfer. Outputs transforms in Freesurfer format .lta for affine and .nii.gz (synthmorph also supports .mgz) for deform, both in RAS orientation. Conversion to other formats is done using lta_convert and mri_warp_convert respectively, which support a wide range of conversion formats and orientations, such as ANTs and FSL. To convert the output of this module, use the registration/convert module successively to it.
```

File: modules/nf-neuro/registration/synthregistration/meta.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+keywords:
+  - registration
```
Comment (AlexVCaron) [inline modules/nf-neuro/registration/synthregistration/meta.yml:5]:
```suggestion
  - Registration
```

Comment (AlexVCaron) [inline modules/nf-neuro/registration/synthregistration/meta.yml:5]: Add relevant keywords for AI/ML

File: modules/nf-neuro/registration/synthregistration/meta.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      type: file
+      description: Nifti volume moving for registration
+      pattern: "*.{nii,nii.gz}"
+
+  - fixed:
+      type: file
+      description: Nifti volume fixed for registration
+      pattern: "*.{nii,nii.gz}"
+
+output:
```
Comment (AlexVCaron) [inline modules/nf-neuro/registration/synthregistration/meta.yml:30]: Order output in the same way as for the module

File: modules/nf-neuro/registration/synthregistration/tests/nextflow.config [hunk 4]
```diff
@@ -0,0 +1,11 @@
+process {
+    withName: "REGISTRATION_SYNTHREGISTRATION" {
+        publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+        memory = 20
```
Comment (AlexVCaron) [inline modules/nf-neuro/registration/synthregistration/tests/nextflow.config:4]: Are you sure you need 20Gb of memory ? I don't remember stating anything over 16Gb, if I remember well ...

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
Again, great work ! Syntax and wording, then merge is on !

File: modules/nf-neuro/registration/synthregistration/tests/nextflow.config (hunk 4 above)
Comment (anroy1) [reply modules/nf-neuro/registration/synthregistration/tests/nextflow.config:4]: You're right. Don't know why i brought it up.

**Review by AlexVCaron — APPROVED** [review]

**Pull request merged**